pkg/splunk/enterprise/util.go

```diff
@@ -48,3 +52,64 @@ func ApplySplunkConfig(client splcommon.ControllerClient, cr splcommon.MetaObjec
 
 	return namespaceScopedSecret, nil
 }
+
+// GetSmartstoreSecrets is used to retrieve S3 access key and secrete keys.
+// To do: sgontla:
+// 1. Support multiple secret objects
+// 2. default secret object
+// volume name other than default, look for the specific secret object, else fetch from defaults
+func GetSmartstoreSecrets(client splcommon.ControllerClient, cr splcommon.MetaObject, smartstore *enterprisev1.SmartStoreSpec) (string, string, error) {
+	var defaultSplunksecret corev1.Secret
+
+	namespacedName := types.NamespacedName{
+		Namespace: cr.GetNamespace(),
+		Name:      GetNamespaceScopedSecretName(cr.GetNamespace()),
+	}
+	err := client.Get(context.TODO(), namespacedName, &defaultSplunksecret)
+	if err != nil {
+		return "", "", err
+	}
+
+	return string(defaultSplunksecret.Data[s3AccessKey]), string(defaultSplunksecret.Data[s3SecretKey]), nil
+}
+
+// CreateSmartStoreConfigMap creates the configMap with Smartstore config in INI format
+func CreateSmartStoreConfigMap(client splcommon.ControllerClient, cr splcommon.MetaObject,
+	smartstore *enterprisev1.SmartStoreSpec) (*corev1.ConfigMap, error) {
+
+	var crKind string
+	crKind = cr.GetObjectKind().GroupVersionKind().Kind
+
+	scopedLog := log.WithName("CreateSmartStoreConfigMap").WithValues("kind", crKind, "name", cr.GetName(), "namespace", cr.GetNamespace())
+
+	if !isSmartstoreConfigured(smartstore) {
+		return nil, fmt.Errorf("Smartstore is not configured")
+	}
+
+	// Get the list of volumes in INI format
+	volumesConfIni, err := GetSmartstoreVolumesConfig(client, cr, smartstore)
```

Comment:
Same as below with len(smartstore.VolList) == 0? Unless these checks are happening elsewhere?

Reply:
This is additional check just to be sure stince the return value is 'string' IndexList and VolList is already checekd as part fo isSmartstoreConfigured()